tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/GitHubActions/GitHubCliCredentialStore.cs

```diff
@@ -0,0 +1,143 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Octokit;
+
+namespace Azure.Sdk.Tools.PipelineWitness.GitHubActions;
+
+public class GitHubCliCredentialStore : ICredentialStore
+{
+    private readonly bool isChainedCredential;
+    private readonly string hostname;
+    private readonly string account;
+    private readonly TimeSpan processTimeout;
+
+    public GitHubCliCredentialStore(string hostname = default, string account = default, TimeSpan? processTimeout = default, bool isChainedCredential = false)
+    {
+        this.processTimeout = processTimeout ?? TimeSpan.FromSeconds(13);
+        this.isChainedCredential = isChainedCredential;
+        this.hostname = hostname;
+        this.account = account;
+    }
+
+    public async Task<Credentials> GetCredentials()
+    {
+        Process process = new()
+        {
+            StartInfo = GetAzureCliProcessStartInfo(),
+            EnableRaisingEvents = true
+        };
+
+        using ProcessRunner processRunner = new(process, this.processTimeout, CancellationToken.None);
+
+        string output;
+        //		try
+        //		{
+        output = await processRunner.RunAsync().ConfigureAwait(false);
+        //		}
+        //		catch (OperationCanceledException)
```

Comment:
Any reason this error handling is commented out?

Reply:
I wasn't really sure how the GH cli or credentials would fail.   I need to test some failure cases and see.

Reply:
I've updated this to remove the commented out error handling.  I'd prefer it fail hard on GH token issues in local dev.  The deployed services will pull the pat from KV and avoid this method